### ISSUE TITLE
Accessibility: Make tooltips hoverable and dismissable with accessibilityLayer prop turned on

### ISSUE THREAD

**Issue by jciruolo** [recharts/recharts#6020]
## Do you want to request a feature or report a bug?

An accessibility feature

## What is the current behavior?

Right now, when a tooltip is hovered via the mouse (even with the accessibilityLayer turned on) the tooltip is not persistent or hoverable since moving the mouse will pass through and highlight the next data point in the graph. Keyboard controls can mitigate this to some degree since we can dismiss by pressing esc or use the keyboard arrow keyes to select the tooltips, but the mouse controls are still impacted by the [WCAG 1.4.13 specification](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html).

## What is the expected behavior?

The expected behavior would be to allow the tooltip to stay open while the user hovers over it, and has the option to be dismissed before a new tooltip would open. This could be a good feature to enable when accessibility layer is turned on, which would allow the tooltips to be accessible for both keyboard AND vision impaired mouse users who may need to move the mouse or change their focus to zoom in on the tooltip to read it per the [specification](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html).

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?

I'm on the latest version and just wanted to say thank you for an excellent library!

**Comment by ckifer** [comment]
>  tooltip is not persistent or hoverable 

Do you mean if the user hovers within the bounds of the tooltip content? Right now, the tooltip may or may not be directly next to the mouse depending on what the user does to adjust it (and animation, etc.) so this would be a difficult change to do correctly across all possible permutations of Tooltip props. Definitely not against it though of course, labeled as a feature request

For now if the graph is changed to use `trigger="click"` then the tooltip will persist at a certain location until dismissed.

---

cc @julianna-langston if you have any thoughts

**Comment by julianna-langston** [comment]
I'm not clear on what the problem is.

If you open a chart with 'accessibilitylayer', and you use a keyboard alone, you can navigate between values. For as long as you are only using the keyboard, the tooltip appears as long as you focus on the relevant value.

If you then start to use a mouse, the tooltip will follow the mouse. If you go back to using the keyboard, then the tooltip follows the keyboard.

All of this behavior is consistent with both WCAG and an intuitive user experience.

I checked out the provided code example, but it doesn't use accessibilityLayer, so I'm still unsure what the problem is.

**Comment by ckifer** [comment]
No example given it looks like, I think that's just the template which hasn't been updated to 3.0 yet (which has accessibilityLayer on by default)

My interpretation is that it's maybe an issue that the user cannot hover on tooltip content when using the mouse without it disappearing?

**Comment by jciruolo** [comment]
I apologize for the example template I left in, I removed it from the issue to reduce confusion if anyone else runs into this in the future.

Per @ckifer's suggestion, I believe changing the tooltip activation to `click` actually resolves the client's concerns around this, I'm glad to know there's already a solution implemented that will keep the tooltip open! Thank you so much for responding and reviewing this.

@julianna-langston, the issue here actually WAS that the tooltip follows the mouse movement. I personally agree with your assessment that the UX is ideal, but it isn't ideal for low-visibility users who need to trigger assistive technology to zoom in on the tooltip, because it typically requires mouse movement to essentially move the magnification zone, causing the tooltip to show different data, especially on graphs with lots of data points. That said, knowing that the trigger="click" option keeps the tooltip open (and we can add some text affordance for users for the keyboard controls) I think this issue can be safely closed with all bases covered here 🎉 

tldr; adding trigger="click" to the Tooltip component, and making sure `accessibilityLayer={true}` has addressed the client's concern 👍

**Comment by ckifer** [comment]
sweet, thanks @jciruolo! closing as resolved
